### PR TITLE
Implement array_sum scalar function

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
## Summary of the changes / Why this improves CrateDB

ArraySum Scalar Function. 

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/expression/scalar/ArraySumFunction.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    @Override
+    public Signature boundSignature() {
+        return boundSignature;
+    }
+
+    @Override
+    public R evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input[] args) {
+        List<T> values = (List) args[0].value();
+        if (values == null || values.isEmpty()) {
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/expression/scalar/ArraySumFunction.java:146]: For most scalars the behavior is that if the incoming value is `null`, the output is also `null`. Not sure if we should have a implicit `null -> 0` conversion here.

We probably also want to have `array_sum` return the same result as `sum()` on an unnested array would return, so if all values of the array are `null` then the result should probably also be null:

```
cr> select sum(x) from unnest([null, null, null]::int[]) as t (x);
+--------+
| sum(x) |
+--------+
|   NULL |
+--------+
SELECT 1 row in set (0.007 sec)
cr> select array_sum([null, null, null]::int[]);
+-----------+
| 0::bigint |
+-----------+
|         0 |
+-----------+
```

File: server/src/main/java/io/crate/expression/scalar/ArraySumFunction.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import java.util.List;
+import java.util.function.Function;
+
+import static io.crate.expression.scalar.array.ArrayArgumentValidators.ensureInnerTypeIsNotUndefined;
+
+public class ArraySumFunction<T extends Number, R extends Number> extends Scalar<R, List<T>> {
+
+    public static final String NAME = "array_sum";
+
+    // ZERO is used in each evaluate call, safe to use across the different rows as Number addition is immutable function.
```
[the remark this answers is not in the document]
Comment (BaurzhanSakhariev) [reply server/src/main/java/io/crate/expression/scalar/ArraySumFunction.java:46]: Yes  - but probably it's obvious and I can remove the comment completely

File: server/src/main/java/io/crate/expression/scalar/ArraySumFunction.java (hunk 1 above)
Comment (BaurzhanSakhariev) [reply server/src/main/java/io/crate/expression/scalar/ArraySumFunction.java:166]: Thx, fixed it and updated test.

**Review by mfussenegger — APPROVED** [review]
:+1:

**Pull request merged**